### PR TITLE
feat: adds outbound proxy env variable support for windows pod

### PR DESCRIPTION
**Pull request by nilekhc**
**Reason for Change**:

This PR adds outbound proxy env variable support for windows pod

**Requirements**

- [x] squashed commits
- [ ] included documentation
- [ ] added unit tests and e2e tests (if applicable).

**Issue Fixed**:

**Does this change contain code from or inspired by another project?**

- [ ] Yes
- [ ] No

**If "Yes," did you notify that project's maintainers and provide attribution?**

**Special Notes for Reviewers**:

### CODE REVIEW

**Comment by nilekhc** [comment]
cc @sozercan

File: manifest_staging/charts/csi-secrets-store-provider-azure/templates/provider-azure-installer-windows.yaml [hunk 1]
```diff
@@ -61,6 +61,13 @@ spec:
             periodSeconds: 30
           resources:
 {{ toYaml .Values.windows.resources | indent 12 }}
+          {{- if .Values.enableArcExtension }}
+          {{- if .Values.Azure.proxySettings.isProxyEnabled }}
```
Comment (aramase) [inline manifest_staging/charts/csi-secrets-store-provider-azure/templates/provider-azure-installer-windows.yaml:65]: We need to add this change too? https://github.com/Azure/secrets-store-csi-driver-provider-azure/blob/e11dd11955657b4a731a0ecfeb3ad6bc2127f723/manifest_staging/charts/csi-secrets-store-provider-azure/templates/provider-azure-installer.yaml#L31-L37

**Review by aramase — COMMENTED** [review]
In addition to the checksum comment, do we need to add this for monitoring for windows provider pods? https://github.com/Azure/secrets-store-csi-driver-provider-azure/blob/e11dd11955657b4a731a0ecfeb3ad6bc2127f723/manifest_staging/charts/csi-secrets-store-provider-azure/templates/provider-azure-installer.yaml#L25-L30

File: manifest_staging/charts/csi-secrets-store-provider-azure/templates/provider-azure-installer-windows.yaml (hunk 1 above)
Comment (nilekhc) [reply manifest_staging/charts/csi-secrets-store-provider-azure/templates/provider-azure-installer-windows.yaml:65]: This is to detect cert file changes. We are not supporting it as a part of this change. So I skipped it.

**Comment by nilekhc** [comment]
> In addition to the checksum comment, do we need to add this for monitoring for windows provider pods?
> 
> https://github.com/Azure/secrets-store-csi-driver-provider-azure/blob/e11dd11955657b4a731a0ecfeb3ad6bc2127f723/manifest_staging/charts/csi-secrets-store-provider-azure/templates/provider-azure-installer.yaml#L25-L30

Added

**Review by aramase — COMMENTED** [review]
`helm lint` is failing with

```bash
[ERROR] templates/: parse error at (csi-secrets-store-provider-azure/templates/provider-azure-installer-windows.yaml:107): unexpected EOF
```

**Review by aramase — APPROVED** [review]

**Pull request merged**